Add new Slick Control Grid Menu (hamburger menu)

ghiscoding opened the pull request:
This a totally new Slick Control to add a Grid Menu (aka hamburger menu on top-right of the grid). This is based on a previous PR #159 (column picker) and also re-used some code and concept from the Header Menu Plugin. 

What does it do?
- Add a fixed/sticky Grid Menu on top-right of the grid
- Possibility to attach the Grid Menu to an external button 
- Possibility to add custom menu items via a simple array provided to the grid options (same concept as the header menu plugin)
  - Once defined, just subscribe to the `onCommand`
- Includes column picker in the grid menu (Column Picker Control is **not** required for this to work, since I copied over the necessary code to produce the column picker).
- Optional title for the custom menu items (top section)
- Optional title for the columns list (bottom section)
- This should work nicely with Touch Devices
- To dismiss the Grid Menu, user can click on the "x" on top right corner or anywhere outside of the Grid Menu

See below for an animated gif of the Grid Menu in action
![2017-09-25_21-58-37](https://user-images.githubusercontent.com/643976/30897118-84505a2c-a322-11e7-9646-be06e71adab2.gif)

ghiscoding commented:
cc @6pac hope you like the new feature :)
I got 2 PRs in the queue #159 and #160, whenever you got a chance to review. 
Thanks 

I want to use that on my next project. 
Also note that, I use NPM to download all files, do I need to update anything for that to work in your next version?

ghiscoding commented:
@6pac any change to see my 2 PRs merged by end of the month? That would be sweet, though I know you are busy, so do it whenever is best for you, by end of month is ok ;)

ghiscoding commented:
still alive @6pac ? I would really need to get this merged, I would like to use it in Production now.

6pac commented:
yeah. sorry. just had about 5 different projects on all in urgent phases. i'll look at it on the weekend (tomorrow). promise.

ghiscoding commented:
Right you're in Australia, it's still Thursday evening in Canada :P
Oh I wish that I could move and work in Australia, been there 2 times and love it. 

Please take the time to enjoy your weekend as well ;)

javastud commented:
Nice feature. Please merge this.

ghiscoding commented:
@6pac thanks a lot for merging. Could you also create a new Release `2.3.9` and also publish on NPM (which is what I use). I cannot use it unless it's publish on NPM.

Thanks again for merging

6pac commented:
Yep. My apologies for not doing those earlier - if I'd realised they didn't modify core code, I would have popped them through straight away. 
I've been trying to knock off a couple of other issues as well, but I'll do the release now. 
Can you clarify how you're using NPM? That is yet another issue that's been hanging around. ie. How does the base repo need to change to accommodate your needs? I started off using an NPM folder with duplicates of the main code files with the NPM headers added, but that's been neglected and I'm not sure anyone's actually using it. Happy to support NPM if I can get a definitive use configuration.

ghiscoding commented:
Doing an NPM publish is as simple as updating your `package.json` file to the new version, run `npm publish` and... that's it. You can see a [video](https://docs.npmjs.com/getting-started/publishing-npm-packages) that explains this process on the NPM site itself. 

As for how I use NPM? I only use it has a way to get the files of your library and unless you make another version, I will never see the new Grid Menu that you just merged. Doing an `npm install slickgrid` is just a way to install packages in 1 simple command (the old way was to download a zip from Github, extract in a vendor folder and use it... but that is so long to do).

So to resume what I'm asking, just update the version in your [package.json](https://github.com/6pac/SlickGrid/blob/master/package.json#L3) to `2.3.9`, save and finally run `npm publish`. That's all :) ... and your probably want to have your Github version to be at the same version, though it's not mandatory, as it removes ambiguity. 

By the way, the `npm` folder has nothing to do with what is actually published on NPM. What matters is actually what is inside the [.npmignore](https://github.com/6pac/SlickGrid/blob/master/.npmignore) file (that tells NPM what is excluded when users run `npm install slickgrid` (you can see that by going in the `node_modules/slickgrid` folder that gets created after running the command. 

If you want to know what I'm doing with Slickgrid at my work, here's my new library for Angular. You can take a look at it [Angular-Slickgrid](https://github.com/ghiscoding/Angular-Slickgrid), there's a demo page too. 

Thanks for the great work and support :)

6pac commented:
OK, all done. Let me know if there's anything I missed.

This has been a productive release. So really, my NPM folder is for NodeJs, not for NPM. All it does is add the EXPORT directives to the top of the file. Anyway, it was not being used by anyone so I've removed it until we get a better system.

ghiscoding commented:
Awesome, I see it all now. Thanks again :)

ghiscoding commented:
@6pac, just so you know. I updated your Wiki to add the Grid Menu in the Examples. Since this is in the "Plugin" folder (the same as the column picker), I renamed the "Plugins" to "Plugins / Controls" inside the Example wiki and added the Grid Menu in there. Feel free to rename if you think it should be different.

6pac commented:
Always happy for collaborative editing. Thanks!
Gotta say, I'd love to get over to Canada sometime too. I lived in the states for a year when I was younger, and going across the border to Canada was like coming home. Everything switched to metric, the streets were clean, ...  I only went to Toronto though, would love to see the North. I've always loved photos of Alaska, and I imagine Canada has a lot of similar country.

ghiscoding commented:
I live in Montreal and let me know if you ever ;)  As for Australia, my best friend moved to Sydney and so I went twice already and will probably go soon again.
